VISION.md

```diff
@@ -0,0 +1,85 @@
+# Vision
+
+A mighty, modern linter and fixer for CSS and CSS-like languages.
+
+One that is:
+
+-   Complete - with coverage of all standard CSS syntax.
+-   Extensible - with multiple points of extension.
+-   Configurable - with no defaults and plenty of options to tailor the linter.
+-   Robust - with comprehensive test coverage and a wide range of fixtures.
+-   Consistent - with conventions for behaviour, naming and documentation.
+-   Performant - with tools to test and improve performance.
+
+## Complete
+
+Provide built-in rules for the following three areas:
+
+-   [Possible errors](docs/user-guide/rules.md#possible-errors).
+-   [Limit language features](docs/user-guide/rules.md#limit-language-features).
+-   [Stylistic issues](docs/user-guide/rules.md#stylistic-issues).
+
+### Possible errors
+
+Provide rules to catch two types of possible errors:
+
+-   Invalid code e.g. invalid hex colors and unknown language features.
+-   Code that is valid but likely has unintended consequences e.g. duplicates and overrides.
+
+Invalid code is best handled by emerging dedicated tools e.g. [csstree](https://github.com/csstree/csstree) - a language parser with syntax validation. As a stop-gap, while these tools mature, provide invalid code rules for the simplest of cases.
+
+(In the future, these tools can be wrapped as plugins to make use of features such as `/* stylelint-* */` command comments, severity levels, and options validator.)
+
+### Limit language features
+
+Provide rules to limit what language features can be used, including:
+
+-   Enforcing a maximum specificity by limiting the specificity itself or the occurrence of different selector types e.g. class, id, attribute etc.
+-   Enforcing best practices _at the configuration level_ e.g. disallowing the `all` keyword for transitions as it is not performant.
+-   Enforcing the use of a subset of features to improve consistency across a code base e.g. limiting what units are allowed (`px` or `rem` etc.)
+-   Enforcing specific patterns for selectors and names (e.g. those of custom properties).
+
+### Stylistic issues
+
+Provide rules to enforce a diverse range of stylistic conventions, including:
+
+-   Whitespace
+-   Case
+-   Quotes
+
+There are two approaches to enforcing stylistic conventions:
+
+-   A machine algorithmically pretty prints the code (usually based on a maximum line length).
+-   A human initially formats the code, and a machine fixes-up/warns-about any mistakes.
+
+The former is handled by pretty printers, like [prettier](https://github.com/prettier/prettier), whereas the latter is catered for by the built-in stylistic rules.
```

Comment:
I think this gives a little too much credit to Prettier. Prettier's lack of configurability is the reason that stylelint is venturing into the `--fix` arena, right? To provide a more comprehensively configurable autofixer? I think it might be worth highlighting the difference between vision for these two types of autofixing. In fact, since I'm sure readers of this document will be very curious about this, I wonder if autofixing warrants a section of its own, to describe the goals particular to that (esp. differences from Prettier)?